### PR TITLE
Give extra left side padding always to tab view

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/30469

Upstream gives extra padding to balance with right side padding space but
it's gone when tab doesn't have sufficient available width.
As we have more narrower left & right padding than upstream, icon seems stick
to left side when extra padding is not used. So use extra padding always.

Default border left inset as icon has left padding.
Screenshot 2023-06-05 at 10 49 21 AM

Increased border left inset as icon doesn't have left padding.
Screenshot 2023-06-05 at 10 49 51 AM

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`BraveTabTest.ExtraPaddingLayoutTest`

### CODE REVIEW

File: browser/ui/views/tabs/brave_tab_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  BraveTabTest() = default;
+  ~BraveTabTest() override = default;
+
+  void LayoutAndCheckBorder(BraveTab* tab,
+                            const gfx::Rect& bounds,
+                            bool gave_extra_padding) {
+    tab->SetBoundsRect(bounds);
+    views::test::RunScheduledLayout(tab);
+    // If we gave extra padding, tab's left border has 12px.
+    // Otherwise upstream's default 8px is.
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab_unittest.cc:25]: delete

File: browser/ui/views/tabs/brave_tab.cc [hunk 2]
```diff
@@ -356,3 +367,21 @@ void BraveTab::AddLayerToBelowThis() {
 
   ReorderChildLayers(layer()->parent());
 }
+
+void BraveTab::UpdateBorder() {
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab.cc:371]: How about overriding UpdateIconVisibility() and setting extra_padding_before_content_ and extra_alert_indicator_padding_ to true? I feel like that would be more compliant with how upstream works.

Comment (simonhong) [reply browser/ui/views/tabs/brave_tab.cc:371]: Actually, I tried that first but it broke layout in some cases like below.
To avoid it, maybe we need to patch more to `Tab::Layout()`.
But increasing border's left inset works well w/o these regressions.

alert icon is clipped(left) compared with right.
Screenshot 2023-06-06 at 9 26 00 AMScreenshot 2023-06-06 at 9 25 49 AM

Icon & alert indicator are too close with x button
Screenshot 2023-06-06 at 9 24 34 AM
Screenshot 2023-06-06 at 9 25 05 AM

Comment (sangwoo108) [reply browser/ui/views/tabs/brave_tab.cc:371]: Fair enough! thanks. Then optionally, I'd like to suggest this routines be moved to GetInsets() instead of setting new border. Setting new border could cause many ops like layout and painting, overriding GetInsets() could be better.

File: browser/ui/views/tabs/brave_tab.h [hunk 3]
```diff
@@ -59,6 +62,8 @@ class BraveTab : public Tab {
   std::unique_ptr<ui::Layer> shadow_layer_;
   gfx::FontList normal_font_;
   gfx::FontList active_tab_font_;
+
+  static constexpr int kExtraLeftPadding = 4;
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab.h:66]: I was hoping we can verify this value is the same value in the tab.cc file, but I can't think of it without touching chromium_src/ overriding.

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: browser/ui/views/tabs/brave_tab.cc (hunk 2 above)
Comment (simonhong) [reply browser/ui/views/tabs/brave_tab.cc:371]: Which `GetInsets()` in `Tab` class?
I think we need to update border based on the result of `UpdateIconVisibility()`.
Fortunately, `UpdateIconVisibility()` is only called at the start of `Layout()` and `UpdateBorder()` is called by `UpdateIconVisibility()`.
So, I think changing border would not cause additional layout?

Comment (simonhong) [reply browser/ui/views/tabs/brave_tab.cc:371]: Ah, do you mean overriding `View::GetInsets()`? Curious how overriding `View::GetInsets()` achieves same goal?

Comment (simonhong) [reply browser/ui/views/tabs/brave_tab.cc:371]: Ah, got it. like below? checking.
```
gfx::Insets BraveTab::GetInsets() const {
  int extra_left_padding = 0;

  // Add extra padding if upstream tab doesn't have it.
  if (!extra_padding_before_content_) {
    extra_left_padding = kExtraLeftPadding;
  }

  auto insets = Tab::GetInsets();
  insets.set_left(insets.left() + extra_left_padding);
  return insets;
}
```

Comment (simonhong) [reply browser/ui/views/tabs/brave_tab.cc]: Ok, updated. Using GetInsets() seems more cleaner. Thanks!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.54.66 Chromium: 114.0.5735.133 (Official Build) nightly (64-bit)
-- | --
Revision | fbfa2ce68d01b2201d8c667c2e73f648a61c4f4a-refs/branch-heads/5735@{#1270}
OS | Windows 11 Version 22H2 (Build 22621.1848)
```

Reproduced the original issue using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/30469#issue-1717685286 running `1.54.38 Chromium: 114.0.5735.90` as per the following:

tabSpacingBrokenWide
tabSpacingBrokenShort

Notice how close the icons are to the left margins as mentioned via the original issue.

Verified that the margins have been increased using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/30469#issue-1717685286 as per the following:

tabSpacingFixedWide
tabSpacingFixedShort

File: browser/ui/views/tabs/BUILD.gn [hunk 4]
```diff
@@ -32,3 +32,23 @@ source_set("browser_tests") {
     ]
   }
 }
+
+source_set("unit_tests") {
+  testonly = true
+
+  sources = [
+    "//chrome/browser/ui/views/tabs/fake_tab_slot_controller.cc",
```
Comment (bridiver) [inline browser/ui/views/tabs/BUILD.gn:40]: why are these included here?

Comment (simonhong) [reply browser/ui/views/tabs/BUILD.gn:40]: Unfortunately, these two files are only included in upstream `unit_tests` target.
